UserDocs.md

```diff
@@ -55,6 +57,8 @@ Writes the in-memory object `src` to the path `dest` in S3.
 Transparently serializes some object types.
 See [Serialization](#serialization) for details.
 
+`dest` should not end in `/`. Objects that end in `/` are treated specially by some s3 tools in a way that is potentially dangerous, so it's best to avoid them.
+
```

Comment:
My general feedback. This bullet belongs in the "Known Issues" section, with a link to the corresponding AWS docs. Does not belong in API docs b/c API should forbid this case whenever possible?